### PR TITLE
drivers: uart: Convert drivers to new DT device macros

### PR DESCRIPTION
**Pull request by galak**
Convert uart drivers from:

	DEVICE_AND_API_INIT -> DEVICE_DT_INST_DEFINE
	DEVICE_GET -> DEVICE_DT_INST_GET
	DEVICE_DECLARE -> DEVICE_DT_INST_DECLARE

etc..

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: drivers/serial/uart_liteuart.c [hunk 1]
```diff
@@ -329,7 +330,7 @@ static int uart_liteuart_init(const struct device *dev)
 
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
 	IRQ_CONNECT(UART_IRQ, DT_INST_IRQ(0, priority),
-			liteuart_uart_irq_handler, DEVICE_GET(uart_liteuart_0),
+			liteuart_uart_irq_handler, DEVICE_DT_INST_GET(uart_liteuart_0),
```
Comment (MaureenHelm) [inline drivers/serial/uart_liteuart.c:333]: The argument to `DEVICE_DT_INST_GET` doesn't look right

File: drivers/serial/uart_native_posix.c [hunk 2]
```diff
@@ -364,14 +364,14 @@ static int np_uart_tty_poll_in(const struct device *dev,
...
-	    DT_INST_LABEL(0), &np_uart_0_init,
+DEVICE_DT_INST_DEFINE(0,
+	    &np_uart_0_init, device_pm_control_nop,
 	    (void *)&native_uart_status_0, NULL,
 	    PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 	    &np_uart_driver_api_0);
 
 #if defined(CONFIG_UART_NATIVE_POSIX_PORT_1_ENABLE)
-DEVICE_AND_API_INIT(uart_native_posix1,
+DEVICE_DT_INST_DEFINE(uart_native_posix1,
```
Comment (henrikbrixandersen) [inline drivers/serial/uart_native_posix.c:374]:
```suggestion
DEVICE_DT_INST_DEFINE(1,
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good apart from one conversion in the native POSIX UART driver, below.

**Review by erwango — APPROVED** [review]

File: drivers/serial/uart_native_posix.c (hunk 2 above)
Comment (galak) [reply drivers/serial/uart_native_posix.c:374]: I'm actually going to leave this alone and revert it back to `DEVICE_AND_API_INIT`.  Need to cleanup the `PORT_1` posix stuff to use device tree instead (but that should be a follow up PR).

**Review by henrikbrixandersen — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**